### PR TITLE
Feat/Health questionnaire

### PR DESCRIPTION
**Pull request by ProggerOleg**
## Describe your changes
- Creted layout for health-questionnaire appointment page
- Created basic slice for this page
- Added saving of state for components, selected values dislayed when user returns to prev step

Screenshot from 2023-11-29 19-39-11
Screenshot from 2023-11-29 19-39-29
Screenshot from 2023-11-29 19-39-47
Screenshot from 2023-11-29 19-40-07
Screenshot from 2023-11-30 09-54-33

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](#https://homecareaid.atlassian.net/jira/software/projects/AC/boards/1?selectedIssue=AC-78)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

### CODE REVIEW

**Comment by nadiia-tsytsylina** [comment]
image

grammar mistake in title. Should be "Challenges"

File: src/components/health-questionnaire/steps/Step3.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+                <Checkbox
+                  checked={selectedOptions.includes(item)}
+                  onChange={(): void => handleOptionSelect(item)}
+                />
+              }
+              label={
+                <Typography
+                  variant="body2"
+                  fontWeight={({ typography }) => typography.fontWeightMedium}
+                  fontSize={15}
```
Comment (Lilly-nn) [inline src/components/health-questionnaire/steps/Step3.tsx:46]: get font size from our theme

**Comment by ProggerOleg** [comment]
> image
> grammar mistake in title. Should be "Challenges"

fixed

File: src/components/health-questionnaire/constants.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  'Operating light switches',
+  'Retrieving and sorting mail',
+  'Pulling cabinets and drawers in kitchen',
+  'Pulling cabinets and drawers in bathroom',
+  'Using oven/stovetop',
+  'Using microwave/toaster/small appliances',
+  'Using large utensils for food preparation',
+  'Entering shower/bathtub',
+  'Opening windows',
+];
```
Comment (lyudad) [inline src/components/health-questionnaire/constants.ts:51]: это не костанты текста которые будут показыватся, сли это текст который будет показыкатся, он должен проходить через файл переводов, иначе когда нужно будет добавлять новый язык начинается трееееш

Comment (ProggerOleg) [reply src/components/health-questionnaire/constants.ts:51]: Got it, fixed

File: src/components/health-questionnaire/steps/SecondStepItem.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+const SecondStepItem = ({
+  questions,
+  options,
+  onCompletion,
+}: {
+  questions: string[];
+  options: string[];
+  onCompletion: (status: boolean) => void;
+}): JSX.Element => {
```
Comment (Simpsonkk) [inline src/components/health-questionnaire/steps/SecondStepItem.tsx:17]: Could you please create type for props before component? as we did for other components

File: src/components/health-questionnaire/steps/Step1.tsx [hunk 4]
```diff
@@ -0,0 +1,84 @@
+import { useState } from 'react';
+import { Checkbox, FormControlLabel, FormGroup, TextField, Typography } from '@mui/material';
+import { useDispatch, useSelector } from 'react-redux';
+import { RootState } from 'src/redux/store';
+import { useLocales } from 'src/locales';
+import { saveNote, selectDiagnosis } from 'src/redux/slices/healthQuestionnaireSlice';
+import { HealthConcernsAndMedicalDiagnoses } from '../constants';
+import { QuestionnaireTypeText, SubmitButton, QuestionnaireContainerContent } from '../styles';
+
+const Step1 = ({ onNext, stepKey }: { onNext: () => void; stepKey: string }): JSX.Element => {
```
Comment (Simpsonkk) [inline src/components/health-questionnaire/steps/Step1.tsx:10]: same

File: src/components/health-questionnaire/steps/Step2.tsx [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+const Step2 = ({
+  onNext,
+  onBack,
+  stepKey,
+}: {
+  onNext: () => void;
+  onBack: () => void;
+  stepKey: string;
+}): JSX.Element => {
```
Comment (Simpsonkk) [inline src/components/health-questionnaire/steps/Step2.tsx:24]: same

File: src/components/health-questionnaire/steps/Step3.tsx [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+
+const Step3 = ({
+  onNext,
+  onBack,
+  stepKey,
+}: {
+  onNext: () => void;
+  onBack: () => void;
+  stepKey: string;
+}): JSX.Element => {
```
Comment (Simpsonkk) [inline src/components/health-questionnaire/steps/Step3.tsx:24]: same

Comment (ProggerOleg) [reply src/components/health-questionnaire/steps/Step3.tsx]: fixed

File: src/components/health-questionnaire/steps/Step2.tsx (hunk 5 above)
Comment (ProggerOleg) [reply src/components/health-questionnaire/steps/Step2.tsx]: fixed

File: src/components/health-questionnaire/index.tsx [hunk 7]
```diff
@@ -0,0 +1,47 @@
+import { useState } from 'react';
+import { useRouter } from 'next/router';
+import Step1 from 'src/components/health-questionnaire/steps/Step1';
+import Step2 from 'src/components/health-questionnaire/steps/Step2';
+import Step3 from 'src/components/health-questionnaire/steps/Step3';
+import { ROUTES } from 'src/routes';
+import { QuestionnaireContainer, Background } from './styles';
```
Comment (nadiia-tsytsylina) [inline src/components/health-questionnaire/index.tsx:7]: better to use absolute imports

File: src/components/health-questionnaire/steps/SecondStepItem.tsx [hunk 8]
```diff
@@ -0,0 +1,73 @@
+import React, { useEffect, useState } from 'react';
+import { Typography } from '@mui/material';
+import { selectActivity } from 'src/redux/slices/healthQuestionnaireSlice';
+import { useDispatch, useSelector } from 'react-redux';
+import { RootState } from 'src/redux/store';
+import { useLocales } from 'src/locales';
+import { ToggleButtonGroupStyled, ToggleButtonStyled } from '../styles';
```
Comment (nadiia-tsytsylina) [inline src/components/health-questionnaire/steps/SecondStepItem.tsx:7]: same here

File: src/components/health-questionnaire/steps/Step1.tsx [hunk 9]
```diff
@@ -0,0 +1,89 @@
+import { useState } from 'react';
+import { Checkbox, FormControlLabel, FormGroup, TextField, Typography } from '@mui/material';
+import { useDispatch, useSelector } from 'react-redux';
+import { RootState } from 'src/redux/store';
+import { useLocales } from 'src/locales';
+import { saveNote, selectDiagnosis } from 'src/redux/slices/healthQuestionnaireSlice';
+import { HealthConcernsAndMedicalDiagnoses } from '../constants';
+import { QuestionnaireTypeText, SubmitButton, QuestionnaireContainerContent } from '../styles';
```
Comment (nadiia-tsytsylina) [inline src/components/health-questionnaire/steps/Step1.tsx:8]: same here

File: src/components/health-questionnaire/steps/Step2.tsx [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+import SecondStepItem from 'src/components/health-questionnaire/steps/SecondStepItem';
+import { RootState } from 'src/redux/store';
+import { saveNote } from 'src/redux/slices/healthQuestionnaireSlice';
+import { Options, ActivitiesOfDailyLivingAssessment } from '../constants';
+import {
+  QuestionnaireContainerContent,
+  QuestionnaireTypeText,
+  CardActionsStyled,
+  ActionButton,
+} from '../styles';
```
Comment (nadiia-tsytsylina) [inline src/components/health-questionnaire/steps/Step2.tsx:14]: same here

File: src/components/health-questionnaire/index.tsx (hunk 7 above)
Comment (ProggerOleg) [reply src/components/health-questionnaire/index.tsx:7]: сильно)), fixed

File: src/components/health-questionnaire/steps/SecondStepItem.tsx (hunk 8 above)
Comment (ProggerOleg) [reply src/components/health-questionnaire/steps/SecondStepItem.tsx:7]: fixed

File: src/components/health-questionnaire/steps/Step1.tsx (hunk 9 above)
Comment (ProggerOleg) [reply src/components/health-questionnaire/steps/Step1.tsx:8]: fixed

**Review by Simpsonkk — APPROVED** [review]

**Pull request merged**